sphinx/writers/html5.py

```diff
@@ -293,10 +293,18 @@ def add_secnumber(self, node):
                 anchorname = '#' + node.parent['ids'][0]
                 if anchorname not in self.builder.secnumbers:
                     anchorname = ''  # try first heading which has no anchor
+
             if self.builder.secnumbers.get(anchorname):
-                numbers = self.builder.secnumbers[anchorname]
-                self.body.append('.'.join(map(str, numbers)) +
-                                 self.secnumber_suffix)
+                return self.builder.secnumbers[anchorname]
+
+        return None
+
+    def add_secnumber(self, node):
+        # type: (nodes.Element) -> None
+        secnumber = self.get_secnumber(node)
+        if secnumber:
+            self.body.append('<span class="section-number">%s</span>' %
```

Comment:
Please update testcases to support this `<span>` tag. It seems all tests are failed now.

About running tests, please read `CONTRIBUTING.rst`.
https://github.com/sphinx-doc/sphinx/blob/master/CONTRIBUTING.rst#getting-started

Reply:
It is not easy to understand tests. It seems the output in case of failure is somehow broken. https://gist.github.com/yves-chevallier/cc699e8b5d259b727242c152efd3b6bb

The following error does not give much information

- What is `[None, None, None, None]`
- There is no `Baz` `into ../../../../../../tmp/pytest-of-ycr/pytest-12/test_build_html_tocdepth/_build/html/index.html`

```
AssertionError: '2.1.1. Baz A' not found in any node matching path .//h4 in index.html: [None, None, None, None]
```